src/KubernetesClient/Kubernetes.List.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.Rest;
+using Microsoft.Rest.Serialization;
+using Newtonsoft.Json;
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace k8s
+{
+    public partial class Kubernetes
+    {        
+        public async Task<HttpOperationResponse<L>> List<T, L>(string namespaceParameter, CancellationToken cancellationToken=default(CancellationToken), bool? watch = default(bool?)) {                                    
```

Comment:
Why do you need this? The List methods are already implemented in the client.

Reply:
Could not find any List method which are generic- takes Type and ListType and gets back a objects of ListType. All of them looked like they are specific - as in uses the Type in the function name and repeats the same code. Am I missing something here ?

Reply:
What I have done elsewhere is pass the List function as a `Func<Type, ListType>` object rather than implementing a generic function like this.